Add semantic field type documentation (#154174)
elastic/elasticsearch: docs/reference/elasticsearch/mapping-reference/field-data-types.md

[`search_as_you_type`](/reference/elasticsearch/mapping-reference/search-as-you-type.md)
:   `text`-like type for as-you-type completion.

[`semantic`](/reference/elasticsearch/mapping-reference/semantic-field.md) {applies_to}`stack: preview 9.5` {applies_to}`serverless: preview`

> :   Generates and searches embeddings for text and multimodal content.

Reviewer: ```suggestion
:   Used for performing semantic search on multimodal content.
```

[`semantic_text`](/reference/elasticsearch/mapping-reference/semantic-text.md)

> :   Used for performing [semantic search](docs-content://solutions/search/semantic-search.md).

Reviewer: ```suggestion
:   Used for performing [semantic search](docs-content://solutions/search/semantic-search.md) on text.
```
